packages/flutter/lib/src/rendering/sliver_grid.dart

```diff
@@ -671,8 +670,7 @@ class RenderSliverGrid extends RenderSliverMultiBoxAdaptor {
       paintExtent: paintExtent,
       maxPaintExtent: estimatedTotalExtent,
       cacheExtent: cacheExtent,
-      // Conservative to avoid complexity.
-      hasVisualOverflow: true,
+      hasVisualOverflow: estimatedTotalExtent > paintExtent || constraints.scrollOffset > 0.0 || constraints.overlap < 0.0,
```

Comment:
Should consideration for overlap work in both directions?
I think overlap is negative when overscrolling the top and positive when overscrolling the bottom.
```suggestion
      hasVisualOverflow: estimatedTotalExtent > paintExtent || constraints.scrollOffset > 0.0 || constraints.overlap != 0.0,
```

Reply:
I only saw it go negative but I probably only tested top. SGTM.